torchdata/dataloader2/reading_service.py

```diff
@@ -304,7 +314,10 @@ def clean_me(process, req_queue, res_queue):
             # TODO(619): Can send terminations simultaneously
             # TODO(620): Make termination a function of QueueWrapperDataPipe (similar to reset)
             req_queue.put(communication.messages.TerminateRequest())
-            _ = res_queue.get()
+            try:
+                _ = res_queue.get(timeout=default_dl2_worker_join_timeout_in_s)
+            except queue.Empty:
+                pass
```

Comment:
I am trying to understand why this was not necessary before? Or was it always needed?

Reply:
I guess it's because there is noting complicated whene `TerminateRquest` is received.
https://github.com/pytorch/data/blob/e15e1453967ce2f25f6fcd2838caadfd0e2fa811/torchdata/dataloader2/communication/iter.py#L149-L151

I felt this `try-except` should be a standard protocol for request-response design.